getdess.sh

```diff
@@ -0,0 +1,280 @@
+#!/bin/bash
+
+# Enable job control
+set -m
+
+# Dess installation
+
+# Dependencies
+# openssl, qrencode, curl
+# docker, docker-compose, snapd
+# certbot (via snap)
+
+# Container Dependencies
+# fuse squashfuse
+
+# SCRIPT GLOBALS
+
+# Supported distros by type
+debian_releases='ubuntu debian'
+redhat_releases='centos fedora'
+
+# Required base packages
+lxc_packages="fuse squashfuse"
+packages="curl openssl qrencode"
+
+# Docker compose link
+compose_url="https://github.com/docker/compose/releases/download/1.29.2/docker-compose"
+
+# Atsign user info
+user_info="atsign, secondaries account, atsign.com"
+
+# Atsign directories
+atsign_dirs="/home/atsign/dess /home/atsign/base /home/atsign/atsign/var /home/atsign/atsign/etc /home/atsign/atsign/logs"
+
+# Repository files
+repo_url="https://raw.githubusercontent.com/atsign-foundation/dess/trunk"
+atsign_files="base/.env base/docker-swarm.yaml base/setup.sh base/shepherd.yaml base/restart.sh"
+dess_scripts="create reshowqr"
+
+# Original user
+original_user=$USER
+
+command_exists () {
+  command -v "$@" > /dev/null 2>&1
+}
+
+# Helper function to check if user's release matches the list
+# $1 = list of valid releases
+# $2 = release to check
+is_release () { [[ $1 =~ (^|[[:space:]])$2($|[[:space:]]) ]]; }
+
+pre_install () {
+  # Get the user's release
+  os_release=$(awk -F= '/^NAME/{print $2}' /etc/os-release | sed 's/\"//g' | awk '{print tolower($1)}')
+  if [ -z "$os_release" ]
+  then
+      echo 'Error: Could not detect your distribution.'
+      exit 0
```

Comment:
Consider non 0 exit codes for failure

Reply:
I put a safe guard in to test against /etc/os-release, some distros need additional packages to run snapd. I will spin up some EC2 instances and do some testing.

Reply:
Made this change